sql/sql_show.cc

```diff
@@ -10619,79 +10619,78 @@ ST_FIELD_INFO check_constraints_fields_info[]=
 };
 
 
+#ifdef HAVE_REPLICATION
 ST_FIELD_INFO slave_status_info[]=
 {
   Column("Connection_name", Name(), NOT_NULL),
   Column("Slave_SQL_State", Varchar(64), NULLABLE),
   Column("Slave_IO_State", Varchar(64), NULLABLE),
   Column("Master_Host", Varchar(HOSTNAME_LENGTH), NULLABLE),
   Column("Master_User", Varchar(USERNAME_LENGTH), NULLABLE),
-  Column("Master_Port", ULong(7), NOT_NULL),
+  Column("Master_Port", Type(&type_handler_ushort, 5, true), NOT_NULL),
   Column("Connect_Retry", SLong(10), NOT_NULL),
   Column("Master_Log_File", Varchar(FN_REFLEN), NOT_NULL),
-  Column("Read_Master_Log_Pos", ULonglong(10), NOT_NULL),
+  Column("Read_Master_Log_Pos", ULonglong(20), NOT_NULL),
   Column("Relay_Log_File", Varchar(FN_REFLEN), NOT_NULL),
-  Column("Relay_Log_Pos", ULonglong(10), NOT_NULL),
+  Column("Relay_Log_Pos", ULonglong(20), NOT_NULL),
   Column("Relay_Master_Log_File", Varchar(FN_REFLEN), NOT_NULL),
-  Column("Slave_IO_Running", Varchar(10), NOT_NULL),
-  Column("Slave_SQL_Running", Varchar(3), NOT_NULL),
-  Column("Replicate_Do_DB", Name(), NOT_NULL),
-  Column("Replicate_Ignore_DB", Name(), NOT_NULL),
-  Column("Replicate_Do_Table", Name(), NOT_NULL),
-  Column("Replicate_Ignore_Table", Name(), NOT_NULL),
-  Column("Replicate_Wild_Do_Table", Name(), NOT_NULL),
-  Column("Replicate_Wild_Ignore_Table", Name(), NOT_NULL),
+  Column("Slave_IO_Running", Varchar(sizeof("Connecting")-1), NOT_NULL),
```

Comment:
I find this unnecessary. Both versions are not future-proof in that a new state with the name length greater than `sizeof("Connecting")-1` might be introduced.
A better approach to me would be to max out the allowed size with `Varchar()` or restore the OLD version plus add a compile-time assert to constrain the string lengths to 10 in [`slave_running[]`](https://github.com/MariaDB/server/blob/mdev-35693/sql/slave.cc#L2861).

I'd go with the simple `Varchar()`.

Reply:
You’re right, both versions are future-proof only by assuming that a new enum is tested to not print truncated in SSS.

Reply:
> […] restore the OLD version plus add a compile-time assert to constrain the string lengths to 10 in `slave_running[]`.

I prefer this better, so these columns aren’t unnecessarily max-sized.
Thus I’ve reverted these changes (at least for now).
This is a 🐞, not a 🐛. Harmless 🐞s should be refactored, not forcibly fixed.